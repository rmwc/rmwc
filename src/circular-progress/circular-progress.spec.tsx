import React from 'react';
import { render } from '@testing-library/react';
import { CircularProgress } from './';

describe('CircularProgress', () => {
  it('renders', () => {
    const { asFragment } = render(<CircularProgress />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('can be sizes', () => {
    const { container, rerender } = render(<CircularProgress size="xsmall" />);
    expect(container.firstChild).toHaveClass(
      'rmwc-circular-progress--size-xsmall'
    );

    rerender(<CircularProgress size="small" />);
    expect(container.firstChild).toHaveClass(
      'rmwc-circular-progress--size-small'
    );

    rerender(<CircularProgress size="medium" />);
    expect(container.firstChild).toHaveClass(
      'rmwc-circular-progress--size-medium'
    );

    rerender(<CircularProgress size="large" />);
    expect(container.firstChild).toHaveClass(
      'rmwc-circular-progress--size-large'
    );

    rerender(<CircularProgress size="xlarge" />);
    expect(container.firstChild).toHaveClass(
      'rmwc-circular-progress--size-xlarge'
    );

    rerender(<CircularProgress size={72} />);
    expect(container).toMatchSnapshot();
  });

  it('can be determinate', () => {
    const { asFragment } = render(<CircularProgress progress={0.3} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('can have a different max / min', () => {
    const { container, rerender } = render(
      <CircularProgress min={0} max={100} progress={30} />
    );
    expect(container.firstChild).toHaveAttribute('aria-valuemax', '100');
    expect(container.firstChild).toHaveAttribute('aria-valuemin', '0');
    expect(container.firstChild).toHaveAttribute('aria-valuenow', '30');

    rerender(<CircularProgress progress={-1} />);
    expect(container.firstChild).toHaveAttribute('aria-valuenow', '-1');

    rerender(<CircularProgress progress={2} />);
    expect(container.firstChild).toHaveAttribute('aria-valuenow', '2');
  });
});
