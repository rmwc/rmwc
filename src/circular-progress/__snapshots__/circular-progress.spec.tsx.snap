// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgress can be determinate 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="1"
    aria-valuemin="0"
    aria-valuenow="0.3"
    class="rmwc-circular-progress"
  >
    <svg
      class="rmwc-circular-progress__circle"
      viewBox="0 0 24 24"
    >
      <circle
        class="rmwc-circular-progress__path"
        cx="12"
        cy="12"
        r="10"
        style="stroke-dasharray: 18.84955592153876, 666.66%;"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`CircularProgress can be sizes 1`] = `
<div>
  <div
    aria-valuemax="1"
    aria-valuemin="0"
    class="rmwc-circular-progress rmwc-circular-progress--indeterminate rmwc-circular-progress--thickerstroke"
    style="font-size: 72px;"
  >
    <svg
      class="rmwc-circular-progress__circle"
      viewBox="0 0 72 72"
    >
      <circle
        class="rmwc-circular-progress__path"
        cx="36"
        cy="36"
        r="30"
      />
    </svg>
  </div>
</div>
`;

exports[`CircularProgress renders 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="1"
    aria-valuemin="0"
    class="rmwc-circular-progress rmwc-circular-progress--indeterminate"
  >
    <svg
      class="rmwc-circular-progress__circle"
      viewBox="0 0 24 24"
    >
      <circle
        class="rmwc-circular-progress__path"
        cx="12"
        cy="12"
        r="10"
      />
    </svg>
  </div>
</DocumentFragment>
`;
